subiquity/common/filesystem/tests/test_manipulator.py

```diff
@@ -91,6 +92,25 @@ def test_delete_encrypted_vg(self):
             a for a in manipulator.model._actions if a.type == 'dm_crypt']
         self.assertEqual(dm_crypts, [])
 
+    def test_wipe_existing_fs(self):
+        # LP: #1983036 - edit a partition to wipe it and mount it, but not
+        # actually change the fs type already there
+        # example: it's ext4, wipe and make it ext4 again
+        manipulator, d = make_manipulator_and_disk()
+        p = make_partition(manipulator.model, d, preserve=True)
+        fs = make_filesystem(manipulator.model, partition=p, fstype='ext2',
+                             preserve=True)
+        manipulator.model._actions.append(fs)
+        manipulator.delete_filesystem(fs)
+        spec = {'wipe': 'random'}
+        with mock.patch.object(p, 'original_fstype') as original_fstype:
+            original_fstype.return_value = 'ext2'
+            fs = manipulator.create_filesystem(p, spec)
+        self.assertFalse(p.preserve)
```

Comment:
This is all very confusing (the gyrations you have to perform in the setup phase of this test is a bad smell!) but I don't think p.preserve should be false after the above sequence of operations? If you take an existing partition and reformat it, the _partition_ shouldn't be destroyed and recreated  -- you would expect the uuid to be preserved, for example. The partition should be wiped though, for sure!

Reply:
Ah yes, the curtin behavior with preserve == True and a wipe value.  I have to admit I found that confusing the first time I came across it as well, I think I just have to accept that this is how curtin works and it's OK.

Fixed!